lib/instrumentation/modules/tedious.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+var semver = require('semver')
+var sqlSummary = require('sql-summary')
+
+var shimmer = require('../shimmer')
+
+module.exports = function (tedious, agent, version, enabled) {
+  if (!enabled) return tedious
+  if (!semver.satisfies(version, '*')) {
+    agent.logger.debug('tedious version %s not supported - aborting...', version)
+    return tedious
+  }
+
+  const ins = agent._instrumentation
+  agent.logger.debug('shimming tedious.Connection')
+  shimmer.wrap(tedious, 'Connection', wrapConnection)
+  shimmer.wrap(tedious, 'Request', wrapRequest)
+
+  return tedious
+
+  function wrapRequest (OriginalRequest) {
+    return class Request extends OriginalRequest {
```

Comment:
It needed the prototype, since it's a constructor, not just a method. Do we have any examples of patching constructors differently? I didn't spot any in a quick search.

The inheritance approach seemed simple and readable to me though.

Reply:
We do it in [generic-pool](https://github.com/elastic/apm-agent-nodejs/blob/12cc4d5617f0928a5727f42bc15967ad94a2c9bb/lib/instrumentation/modules/generic-pool.js#L17-L20) but it ain't pretty. So your approach is a lot cleaner 👍